zerver/lib/actions.py

```diff
@@ -2263,6 +2265,38 @@ def check_send_typing_notification(sender: UserProfile, user_ids: List[int], ope
     )
 
 
+def do_send_stream_typing_notification(
+    realm: Realm, sender: UserProfile, operator: str, stream_id: int, topic: str
+) -> None:
+
+    sender_dict = {"user_id": sender.id, "email": sender.email}
+
+    event = dict(
+        type="stream_typing",
+        op=operator,
+        sender=sender_dict,
+        stream_id=stream_id,
+        topic=topic,
+    )
+
+    # Only deliver the notification to active users subscribed to the stream
+    stream_subscriptions = get_active_subscriptions_for_stream_id(stream_id)
+    users_subscribed = [subscription.user_profile for subscription in stream_subscriptions]
+    user_ids_to_notify = [user.id for user in users_subscribed]
+
+    send_event(realm, event, user_ids_to_notify)
+
+
+def check_send_stream_typing_notification(
+    sender: UserProfile, operator: str, stream_id: int, topic: str
+) -> None:
+    realm = sender.realm
+
+    access_stream_by_id(sender, stream_id)  # Does the checking
+
+    do_send_stream_typing_notification(realm, sender, operator, stream_id, topic)
```

Comment:
I don't think we need this function -- we should just do `access_stream_by_id` in the views code (as is our usual policy) and then pass the `Stream` object to `do_send_stream_typing_notification`. (which can then just access `stream.id`).

Reply:
We don't need a `Stream` object anywhere. Moving that `access_stream_by_id` check to the view function and removing this function.